packages/shared/src/hooks/post/useBlockPostPanel.ts

```diff
@@ -0,0 +1,273 @@
+    initialData: {},
+    ...disabledRefetch,
+  });
+  const setShowTagsPanel = useCallback(
+    (params: BlockData) =>
+      client.setQueryData<BlockData>(key, (current) => ({
+        ...current,
+        ...params,
+      })),
+    [client, key],
+  );
+  const { reportPost } = useReportPost();
+  const { trackEvent } = useAnalyticsContext();
+
+  const value = useMemo<BlockData>(
+    () => ({
+      ...data,
+      showTagsPanel: checkHasCompleted(ActionType.HideBlockPanel)
+        ? undefined
+        : data?.showTagsPanel,
+    }),
+    [data, checkHasCompleted],
+  );
+
+  const updateFeedPreferences = useCallback(
+    async (
+      blocks: string[],
+      unblocks: string[],
+      shouldBlockSource?: boolean,
+    ) => {
+      const onUpdateSource = shouldBlockSource
+        ? onUnfollowSource
+        : onFollowSource;
+
+      const results = await Promise.all([
+        blocks.length ? onBlockTags({ tags: blocks }) : ignoredCall(),
+        unblocks.length ? onUnblockTags({ tags: unblocks }) : ignoredCall(),
+        isNullOrUndefined(shouldBlockSource)
+          ? ignoredCall()
+          : onUpdateSource({ source: post.source }),
+      ]);
+
+      return results.every(({ successful }) => successful);
+    },
+    [onFollowSource, onUnfollowSource, onBlockTags, onUnblockTags, post],
+  );
+
+  const onUndo = useCallback(
+    async (blocks: string[], shouldBlockSource?: boolean) => {
+      const successful = await updateFeedPreferences(
+        [],
+        blocks,
+        shouldBlockSource,
+      );
+
+      if (!successful) return;
+
+      setShowTagsPanel({
+        showTagsPanel: undefined,
+        blocked: {},
+      });
+    },
+    [setShowTagsPanel, updateFeedPreferences],
+  );
+
+  const onClose = useCallback(
+    (forceClose?: boolean) =>
+      setShowTagsPanel({ showTagsPanel: forceClose ? undefined : false }),
+    [setShowTagsPanel],
+  );
+
+  const onShowPanel = useCallback(
+    () => setShowTagsPanel({ showTagsPanel: true }),
+    [setShowTagsPanel],
+  );
+
+  const onDismissPermanently = useCallback(() => {
+    completeAction(ActionType.HideBlockPanel);
+    setShowTagsPanel({ showTagsPanel: undefined });
+  }, [completeAction, setShowTagsPanel]);
+
+  const onReport = useCallback(() => {
+    openModal({
+      type: LazyModal.ReportPost,
+      props: {
+        post,
+        onReport: async (
+          reportPostIndex,
+          reportedPost,
+          reason,
+          comment,
+          blockSource,
+          tags,
+        ): Promise<void> => {
+          const { successful } = await reportPost({
+            id: reportedPost?.id,
+            reason,
+            comment,
+            tags,
+          });
+
+          if (!successful) {
+            return;
+          }
+
+          trackEvent(
+            postAnalyticsEvent('report post', reportedPost, {
+              extra: { origin: Origin.TagsFilter },
+            }),
+          );
+        },
+        postIndex: -1,
```

Comment:
This feels a bit weird to me, can we not make it optional and not send it if it doesn't exist?
We don't use it for this case so it feels very weird.

Do we even need it for the old flow?